airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py

```diff
@@ -1031,25 +1037,26 @@ def stream_slices(
         self.set_sync(sync_mode)
         chunk_size = pendulum.duration(days=30)
         slices = []
+        if self._start_date:
+            now_ts = int(pendulum.now().timestamp() * 1000)
+            start_ts = int(self._start_date.timestamp() * 1000)
+            max_delta = now_ts - start_ts
+            chunk_size = int(chunk_size.total_seconds() * 1000) if self.need_chunk else max_delta
+
+            for ts in range(start_ts, now_ts, chunk_size):
+                end_ts = ts + chunk_size
+                slices.append(
+                    {
+                        "startTimestamp": ts,
+                        "endTimestamp": end_ts,
+                    }
+                )
+            # Save the last slice to ensure we save the lastest state as the initial sync date
+            if len(slices) > 0:
+                self.last_slice = slices[-1]
 
-        now_ts = int(pendulum.now().timestamp() * 1000)
-        start_ts = int(self._start_date.timestamp() * 1000)
-        max_delta = now_ts - start_ts
-        chunk_size = int(chunk_size.total_seconds() * 1000) if self.need_chunk else max_delta
-
-        for ts in range(start_ts, now_ts, chunk_size):
-            end_ts = ts + chunk_size
-            slices.append(
-                {
-                    "startTimestamp": ts,
-                    "endTimestamp": end_ts,
-                }
-            )
-        # Save the last slice to ensure we save the lastest state as the initial sync date
-        if len(slices) > 0:
-            self.last_slice = slices[-1]
-
-        return slices
+            return slices
+        return [None]
```

Comment:
This makes me a little bit nervous to regress functionality back to basically having just one slice for the entire stream if the user doesn't specify a `start_time`. And because it's now optional, this might start a trend of a lot of customers not choosing a start date potentially.

Reply:
I agree with @brianjlai there. Can we add a logic which would determine a start date according to early record consumption or by querying an endpoint which would return the hubspot account creation date?
If this is not possible can we artificially set a start date to june 2006 (when hubspot was created)?